File: server/Netptune.Core/Requests/UpdateWorkspaceRequest.cs
using System.ComponentModel.DataAnnotations;

using Netptune.Core.Meta;

namespace Netptune.Core.Requests;

public record UpdateWorkspaceRequest
{
    [Required]
    public string? Slug { get; init; }

    public string? NewSlug { get; init; }

    public string? Name { get; init; }

    public string? Description { get; init; }

    public WorkspaceMeta? MetaInfo { get; init; }

    public bool? IsPublic { get; init; }

    public bool? AssistantEnabled { get; init; }

    public bool? AllowAssistantDataSampling { get; init; }

    public List<string>? PublicPermissions { get; init; }

    public long? MaxUploadBytes { get; init; }
}

File: client/src/app/core/models/requests/update-workspace-request.ts
import { Permission } from '@core/auth/permissions';
import { WorkspaceMeta } from '../workspace';

export interface UpdateWorkspaceRequest {
  slug: string;
  newSlug?: string;
  name?: string;
  description?: string;
  metaInfo: WorkspaceMeta;
  isPublic?: boolean;
  assistantEnabled?: boolean;
  allowAssistantDataSampling?: boolean;
  publicPermissions?: Permission[];
  maxUploadBytes?: number;
}
